department-of-veterans-affairs/caseflow
# Fetch claimant information from BGS

Connects #5632.

Adds attributes for Claimant model using information fetched from BGS, and returns those attributes in the `WorkQueue::AppealSerializer` if a claimant exists for the appeal.

### Testing Plan
1. Deploy branch to UAT
2. Look for test Veteran in [test data](https://github.com/department-of-veterans-affairs/appeals-qa/blob/master/TestData/TestCertificationData.md), find one that looks promising and make a request to BGS to get the participant ID
```
rails> u = User.first
rails> RequestStore[:current_user] = u
rails> bgs = ExternalApi::BGSService.new
rails> bgs.fetch_poa_by_file_number(id) # For C file numbers (no trailing C)
rails> bgs.fetch_file_number_by_ssn(id) # For SSNs
```
3. Check to see if we fetch data from BGS properly
```
rails> u = User.first
rails> RequestStore[:current_user] = u
rails> participant_id = {{12345}}
rails> c = Claimant.create(review_request_type: "review_request_type", review_request_id: 1, participant_id: participant_id)
rails> c.address_line_1 # Kicks off BGS fetch and returns some value
rails> c.relationship # Will be null because BGS UAT sandbox does not have data here. Works in prod.
```
